File: DesignPatternsExercises/ruby/composite_design_pattern/solution/composite_design_pattern.rb
require_relative 'block'
require_relative 'group'
require_relative 'structure'

block1 = Block.new(1, 2)
block2 = Block.new(3, 4)
block3 = Block.new(5, 6)

parent_structure = Structure.new(:parent)
child_structure1 = Structure.new(:child1)
child_structure2 = Structure.new(:child2)

child_structure1.add(block1)
child_structure1.add(block2)
child_structure2.add(block3)

parent_structure.add(child_structure1)
parent_structure.add(child_structure2)

# parent_structure.__________(1)__________
parent_structure.assemble

File: DesignPatternsExercises/java/CompositeDesignPattern/Exercise/CompositeDesignPattern.java
public class CompositeDesignPattern {
	public static void main(String[] args) {
		Block block1 = new Block(1,2);
		Block block2 = new Block(3,4);
		Block block3 = new Block(5,6);

		Structure parentStructure = new Structure("parent");
		Structure childStructure1 = new Structure("child1");
		Structure childStructure2 = new Structure("child2");

		childStructure1.add(block1);
		childStructure1.add(block2);
		childStructure2.add(block3);

		parentStructure.add(childStructure1);
		parentStructure.add(childStructure2);
		// parentStructure.__________(1)__________();
		parentStructure.assemble();
	}
}
